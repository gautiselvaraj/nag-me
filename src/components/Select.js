import React from 'react';
import styled from 'styled-components';
import Label from './Label';

const SelectWrap = styled.div`
  display: flex;
  flex-direction: ${props => props.inline ? 'row' : 'column'};
`;

const Select = styled.select`
  font-size: ${props => props.small ? '.85rem' : '1rem'};
  height: ${props => props.small ? '1.5rem' : '2rem'};
  margin-top: 0;
  margin-bottom: 0;
  ${props => props.inline ? '' : 'width: 100%;'};
`;

export default ({children, label, id, ...otherProps}) => (
  <SelectWrap {...otherProps}>
    {!!label &&
      <Label htmlFor={id} {...otherProps}>{label}</Label>
    }
    <Select id={id} {...otherProps}>
      {children}
    </Select>
  </SelectWrap>
)
