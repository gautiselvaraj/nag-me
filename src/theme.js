export default {
  main:           '#214154',
  mainDarken:     '#083148',
  mainLighten:    '#215a7b',
  black:          '#111111',
  white:          '#f6f6f6',
  greyLightest:   '#efefef',
  greyLighter:    '#dddddd',
  greyLight:      '#a5a5a5',
  grey:           '#898989',
  greyDark:       '#777777',
  greyDarker:     '#595959',
  greyDarkest:    '#222',
}
