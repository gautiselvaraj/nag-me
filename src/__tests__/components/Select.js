import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Select from '../../components/Select';
import theme from '../../theme';
import 'jest-styled-components';

describe('<Select />', () => {
  describe('without error and label', () => {
    it('should match snapshot', () => {
      const tree = renderer
        .create(
          <Select theme={theme}>
            <option>Option 1</option>
          </Select>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have correct element, text, height & font-size based on small prop', () => {
      let select = mount(
        <Select theme={theme}>
          <option>Option 1</option>
        </Select>
      );
      expect(select.find('select').length).toBe(1);
      expect(select.find('select option').text()).toBe('Option 1');
      expect(select.find('select')).toHaveStyleRule('font-size', '1rem');
      expect(select.find('select')).toHaveStyleRule('height', '2rem');

      select = mount(
        <Select small theme={theme}>
          <option>Option 1</option>
        </Select>
      );
      expect(select.find('select')).toHaveStyleRule('font-size', '.85rem');
      expect(select.find('select')).toHaveStyleRule('height', '1.5rem');
    });
  });

  describe('with error and label', () => {
    it('should match snapshot', () => {
      const tree = renderer
        .create(
          <Select label="Label 1" error="Error 1" theme={theme}>
            <option>Option 1</option>
          </Select>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have label element if passed', () => {
      const select = mount(
        <Select label="Label 1" error="Error 1" theme={theme}>
          <option>Option 1</option>
        </Select>
      );
      expect(select.find('label').length).toBe(1);
      expect(select.find('label').text()).toBe('Label 1');
    });
  });
});
