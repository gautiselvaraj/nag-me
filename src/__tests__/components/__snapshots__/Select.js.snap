// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> with error and label should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  border: 1px solid #dddddd;
  box-sizing: border-box;
  font-size: 1rem;
  height: 2rem;
  margin-top: 0;
  margin-bottom: 0;
  width: 100%;
}

.c1 {
  font-size: 1rem;
  line-height: 2rem;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor={undefined}
  >
    Label 1
  </label>
  <select
    className="c2"
    id={undefined}
  >
    <option>
      Option 1
    </option>
  </select>
</div>
`;

exports[`<Select /> without error and label should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  border: 1px solid #dddddd;
  box-sizing: border-box;
  font-size: 1rem;
  height: 2rem;
  margin-top: 0;
  margin-bottom: 0;
  width: 100%;
}

<div
  className="c0"
>
  <select
    className="c1"
    id={undefined}
  >
    <option>
      Option 1
    </option>
  </select>
</div>
`;
